### PR TITLE
Add API for loading specific nuget.config in ReferenceAssemblies

### PR DESCRIPTION
**Pull request by KatoStoelen**
This opt-in feature enables use of a specific nuget.config file in analyzer tests, instead of machine and user wide settings. When specifying a path to a nuget.config file, only that file is loaded.

Closes: #551

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn-sdk?pullRequest=639) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn-sdk?pullRequest=639) before we can accept your contribution.<br/><br/>:x: KatoStoelen [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn-sdk?pullRequest=639)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn-sdk?pullRequest=639) it.</sub>

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs [hunk 1]
```diff
@@ -151,11 +155,14 @@ public ReferenceAssemblies AddLanguageSpecificAssemblies(string language, Immuta
         }
 
         public ReferenceAssemblies WithPackages(ImmutableArray<PackageIdentity> packages)
-            => new ReferenceAssemblies(TargetFramework, AssemblyIdentityComparer, ReferenceAssemblyPackage, ReferenceAssemblyPath, Assemblies, FacadeAssemblies, LanguageSpecificAssemblies, packages);
+            => new ReferenceAssemblies(TargetFramework, AssemblyIdentityComparer, ReferenceAssemblyPackage, ReferenceAssemblyPath, Assemblies, FacadeAssemblies, LanguageSpecificAssemblies, packages, NuGetConfigFilePath);
 
         public ReferenceAssemblies AddPackages(ImmutableArray<PackageIdentity> packages)
             => WithPackages(Packages.AddRange(packages));
 
+        public ReferenceAssemblies WithNuGetConfigFilePath(string nugetConfigFilePath)
```
Comment (KatoStoelen) [inline src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs:163]: Maybe `UsingNuGetConfigFile(...)` would be a better name?

Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs:163]: The current `With` prefix is consistent with the way we name things

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
Thanks @KatoStoelen! (and sorry for the delay!)

**Pull request merged**